### ISSUE TITLE
How to use other identity loss rather than Arcface or Moco

### ISSUE THREAD

**Issue by mmitchef** [eladrich/pixel2style2pixel#319]
Hello, 

Thanks for sharing your code. I have a pre-trained stylegan model on my custom dataset (human eyes). This dataset consists of pairs of eyes and I need to preserve the identity loss. _I can not use Arcface or Moco identity loss for this purpose._ 

I have a code that calculate the identity loss for eyes and I want to include it into your code. But, I am confused how to change your code and adopt it for my purpose. **Can I still use ResNet backbone to extract the feature maps on my dataset domain and replace my identity loss with your at**
`"./criteria/id_loss.py".`

I guess I need to change lines **49-50, 211-214** of `training/coach.py` based on my identity loss.

Would you provide me some information how I can get your code to work for my purpose.
 
Thanks for taking time responding me.

**Comment by yuval-alaluf** [comment]
You're correct. You'll want to change the function `calc_loss` to add your custom loss. This function will receive the original input image and the reconstructed image and you can pass these images to your custom loss function where you'll be able to compute any loss you want over the two images.
You'll still be able to use the ResNet backbone since the encoder is independent from the loss functions used during training.

**Comment by mmitchef** [comment]
Thank you,
